rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -123,6 +125,9 @@
   private Counter cacheRequestCount;
   private Counter nonlocalRequestCount;
 
+  // maintain generation number for remote file
+  protected static ConcurrentHashMap<String, Integer> generationNumber = new ConcurrentHashMap<>();
```

Comment:
memory is leaked on this map. This can cause big problems in long lived deployments that access millions of files over time